dask/dataframe/io/io.py

```diff
@@ -646,24 +652,25 @@ def to_records(df):
     return df.map_partitions(M.to_records)
 
 
-# TODO: type this -- causes lots of papercuts
 @insert_meta_param_description
 def from_delayed(
-    dfs,
+    dfs: Delayed | distributed.Future | Iterable[Delayed | distributed.Future],
     meta=None,
-    divisions=None,
-    prefix="from-delayed",
-    verify_meta=True,
-):
+    divisions: tuple[Any, ...] | Literal["sorted"] | None = None,
+    prefix: str = "from-delayed",
+    verify_meta: bool = True,
+) -> DataFrame:
```

Comment:
Super-low-priority comment -- I think both `from_delayed` and `from_pandas` can return a Dask DataFrame **Series** object as well, should we include that in the type check?

Reply:
Oh, that's interesting. I guess that wasn't caught by mypy because `new_dd_object` isn't yet typed (although I might have typed it in my other PR). Happy to change this.